tests/api/route-services/samples.test.js

```diff
@@ -149,4 +148,58 @@ describe('tests for the samples service', () => {
       })
       .then(() => done());
   });
+
+  it('addSample works', (done) => {
+    const mockSampleToAdd = {
+      name: 'WT1',
+      projectUuid: 'projectUuid',
+      uuid: 'sampleUuid',
+      type: '10X Chromium',
+      species: null,
+      createdDate: '4242-42-42T18:58:49.131Z',
+      lastModified: '4242-42-42T18:58:49.131Z',
+      complete: false,
+      error: false,
+      fileNames: [],
+      files: {},
+      metadata: {},
+    };
+
+    const updateItemSpy = mockDynamoUpdateItem({});
+
+    const expectedProjectsUpdate = {
+      TableName: 'projects-test',
+      Key: { projectUuid: { S: 'projectUuid' } },
+      UpdateExpression: 'SET projects.samples = list_append(projects.samples, :newSampleId)',
+      ExpressionAttributeValues: AWS.DynamoDB.Converter.marshall({ ':newSampleId': ['sampleUuid'] }),
+      ReturnValues: 'UPDATED_NEW',
+    };
+
+    const expectedSamplesUpdate = {
+      TableName: 'samples-test',
+      Key: {
+        experimentId: {
+          S: 'experimentId',
+        },
+      },
+      UpdateExpression: 'SET samples.#newSampleId = :newSample, projectUuid = :projectUuid',
+      ExpressionAttributeNames: {
+        '#newSampleId': 'sampleUuid',
+      },
+      ExpressionAttributeValues: AWS.DynamoDB.Converter.marshall({
+        ':newSample': mockSampleToAdd,
+        ':projectUuid': 'projectUuid',
+      }),
+      ReturnValues: 'UPDATED_NEW',
+    };
+
+    (new SamplesService()).addSample('projectUuid', 'experimentId', mockSampleToAdd)
+      .then((res) => {
+        expect(res).toEqual(OK());
+
+        expect(updateItemSpy.mock.calls).toContainEqual([expectedSamplesUpdate]);
+        expect(updateItemSpy.mock.calls).toContainEqual([expectedProjectsUpdate]);
+      })
+      .then(() => done());
+  });
```

Comment:
Likewise as fo the projects, if possible change this test function body into `async` and break this code open.